homeassistant/components/omnilogic/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow for Omnilogic integration."""
+import logging
+
+from omnilogic import LoginException, OmniLogic
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN, POLL_INTERVAL  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Omnilogic."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        config_entry = self.hass.config_entries.async_entries(DOMAIN)
+        if config_entry:
+            return self.async_abort(reason="already_setup")
+
+        if user_input is not None:
+
+            username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
+
+            try:
+                omni = OmniLogic(username, password)
+                await omni.connect()
+            except LoginException:
+                errors["base"] = "cannot_connect"
+            except Exception:
```

Comment:
I think you're missing a test case for this

Reply:
There is handling for both cannot_connect (line 93-101 in the test_config_flow) and general Exception (line 103-107 in the test_config_flow).

Which other case do you see missing?

Reply:
Oh, I see. The test name is confusing, I think you should either split into two tests, or change the name.